models/classes/TestCategoryPresetProvider.php

```diff
@@ -86,6 +87,65 @@ public function getPresets() {
         return $this->allPresets;
     }
 
+    /**
+     * Get all active presets matching the given config.
+     *
+     * If a preset is linked to a feature flag,
+     * we add it only if the config value matching the flag is true.
+     *
+     * For example, if a $aPreset->featureFlag = 'foo';
+     * The preset will be included only if $config['foo'] = true.
+     *
+     * If the config doesn't have a flag, we keep the preset.
+     *
+     * @param array $config a config flag list as  { key : string => value : boolean }
+     * @return array the sorted preset list
+     */
+    public function getAvailablePresets($config = []) {
+        //work on a clone
+        $presets = array_merge([], $this->getPresets());
+
+        foreach($presets as $groupId => &$presetGroup) {
+
+            if(isset($presetGroup['presets'])){
+
+                //filter presets based on the config value
+                //if the config has the flag, we chek it's value
+                //if the config doesn't have the flag, we keep the preset
+                $presetGroup['presets'] = array_filter(
+                    $presetGroup['presets'],
+                    function($preset) use ($config) {
+                        return !$this->isPresetAvailable($preset, $config);
+                    }
+                );
+
+                //remove empty groups
+                if(count($presetGroup['presets']) === 0){
+                    unset($presets[$groupId]);
+                }
+            }
+        }
+        return $presets;
+    }
+
+    /**
+     * Is a preset available according to a configuration (ie. based on it's featureFlag)
+     *
+     * @param TestCategoryPreset $preset the preset to test 
+     * @param array $config the configuration
+     * @return boolean true if available
+     */
+    private function isPresetAvailable(TestCategoryPreset $preset, $config = [])
+    {
+        if(!is_null($preset)) {
+            $flag = $preset->getFeatureFlag();
+            if(is_string($flag) && isset($config[$flag]) && $config[$flag] != true){
```

Comment:
Instead of `is_string($flag)` you can use only `$flag` to check whether it has a value or not.